itest/rust/src/array_test.rs

```diff
@@ -223,75 +257,251 @@ fn array_min_max() {
+    assert_eq!(array.pop(), Some(2));
+    assert_eq!(array.pop_front(), Some(1));
 
     assert_eq!(array.pop(), None);
     assert_eq!(array.pop_front(), None);
 }
 
 #[itest]
 fn array_insert() {
-    let mut array = Array::from(&[1, 2]);
+    let mut array = array![1, 2];
 
-    array.insert(0, 3.to_variant());
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![3, 1, 2]);
+    array.insert(0, 3);
+    assert_eq!(array, array![3, 1, 2]);
 
-    array.insert(3, 4.to_variant());
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![3, 1, 2, 4]);
+    array.insert(3, 4);
+    assert_eq!(array, array![3, 1, 2, 4]);
 }
 
 #[itest]
 fn array_extend() {
-    let mut array = Array::from(&[1, 2]);
-    let other = Array::from(&[3, 4]);
+    let mut array = array![1, 2];
+    let other = array![3, 4];
     array.extend_array(other);
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![1, 2, 3, 4]);
+    assert_eq!(array, array![1, 2, 3, 4]);
 }
 
 #[itest]
 fn array_sort() {
-    let mut array = Array::from(&[2, 1]);
+    let mut array = array![2, 1];
     array.sort_unstable();
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![1, 2]);
+    assert_eq!(array, array![1, 2]);
 }
 
 #[itest]
 fn array_reverse() {
-    let mut array = Array::from(&[1, 2]);
+    let mut array = array![1, 2];
     array.reverse();
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![2, 1]);
+    assert_eq!(array, array![2, 1]);
 }
 
 #[itest]
 fn array_shuffle() {
-    // Since the output is random, we just test that it doesn't crash.
-    let mut array = Array::from(&[1i64]);
+    let mut array = array![1];
     array.shuffle();
-    assert_eq!(array.try_to_vec::<i64>().unwrap(), vec![1]);
+    assert_eq!(array, array![1]);
+}
+
+#[itest]
+fn array_mixed_values() {
+    let int = 1;
+    let string = GodotString::from("hello");
+    let packed_array = PackedByteArray::from(&[1, 2]);
+    let typed_array = array![1, 2];
+    let object = Object::new_alloc();
+    let node = Node::new_alloc();
+    let ref_counted = RefCounted::new();
+
+    let array = array![
+        int.to_variant(),
+        string.to_variant(),
+        packed_array.to_variant(),
+        typed_array.to_variant(),
+        object.to_variant(),
+        node.to_variant(),
+        ref_counted.to_variant(),
+    ];
+
+    assert_eq!(i64::try_from_variant(&array.get(0)).unwrap(), int);
+    assert_eq!(
+        GodotString::try_from_variant(&array.get(1)).unwrap(),
+        string
+    );
+    assert_eq!(
+        PackedByteArray::try_from_variant(&array.get(2))
+            .unwrap()
+            .len(),
+        packed_array.len()
+    ); // TODO Use PackedByteArray Eq impl once available
+    assert_eq!(
+        TypedArray::try_from_variant(&array.get(3)).unwrap(),
+        typed_array
+    );
+    assert_eq!(
+        Gd::<Object>::try_from_variant(&array.get(4))
+            .unwrap()
+            .instance_id_or_none(),
+        object.instance_id_or_none()
```

Comment:
This would theoretically pass if both instance IDs are 0 (`None`).

A stronger test would be to call `instance_id()` directly, as that ensures both sides have a valid instance ID. As a nice bonus, it's also more terse.

Reply:
Yes, that's better, don't know what I was thinking.